sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala

```diff
@@ -343,6 +343,15 @@ case class ShowTableStatement(
     partitionSpec: Option[TablePartitionSpec])
   extends ParsedStatement
 
+/**
+ * A SHOW VIEWS statement, as parsed from SQL.
+ */
+case class ShowViewsStatement(
```

Comment:
can we follow `ShowTables` and name it `ShowViews`?

Reply:
A bit more background: SHOW TABLES has 2 versions: v1 command (`ShowTablesCommand`) and v2 command (`ShowTables`). That's why we need `ShowTablesStatement`, as we need to check some condition and convert it to either v1 or v2 command.

SHOW VIEWS does not need a v1 command, so we only need a single v2 command
```
case class ShowViews(namespace: LogicalPlan, pattern: Option[String]) ...
```
and its corresponding physical plan `ShowViewsExec`.

BTW we need to follow `ShowTables` and special-case the current namespace in the rule `ResolveNamespace`

Reply:
Thanks so much for the background introduction! I think I misunderstood the scope between V2 command and V2 SessionCatalog, that's why I wrongly followed V1 command style before. 
Sure, I'll update it to V2 command.

Reply:
@cloud-fan I did the implementation and got two more questions:
1. With the v2 command style, we need to extend the `CatalogPlugin` to support view related interface. Which way do you prefer, add listViews function to `TableCatalog` or add new interface `SupportViews`? 
2. The current `CatalogPlugin` also does not support `isTemporaryTable` check, so the v2 `ShowTablesExec` and `ShowViewsExec` can not support the isTemporary output column. Do we need to handle it here?

Directly calling `SessionCatalog` API instead of  'CatalogPlugin' seems not elegant here, right?

Reply:
Let's not add the view APIs for now, as it needs more design work. And seems like we need a v1 version as there is no view in v2 api.

So my suggestion is:
1. have the v2 style `ShowViews`, without the physical plan as v2 catalog doesn't have view
2. in `ResolveSessionCatalog`, catch `ShowViews` and fail if the catalog is not a session catalog. Otherwise, convert it v1 style `ShowViewsCommand`.

Reply:
Thanks for the suggestions, updated in 90d70196de70b7a93b01e188fda85682a65dc24e using v2 style `ShowViews`.